extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java

```diff
@@ -234,6 +236,33 @@ public void openshiftRequirementsNative(OpenshiftConfig openshiftConfig,
         }
     }
 
+    @BuildStep(onlyIf = { OpenshiftBuild.class })
+    public void configureExternalRegistry(ApplicationInfoBuildItem applicationInfo,
+            OpenshiftConfig openshiftConfig,
+            ContainerImageInfoBuildItem containerImageInfo,
+            BuildProducer<DecoratorBuildItem> decorator) {
+        containerImageInfo.registry.ifPresent(registry -> {
+            String imagePushSecret = openshiftConfig.imagePushSecret.orElse(applicationInfo.getName() + "-push-secret");
+            if (openshiftConfig.imagePushSecret.isPresent()) {
+                // Process
+            } else if (containerImageInfo.username.isPresent() && containerImageInfo.password.isPresent()) {
+                final String name = applicationInfo.getName() + "-push-secret";
+                decorator.produce(new DecoratorBuildItem(OPENSHIFT,
+                        new AddDockerConfigJsonSecretDecorator(name, containerImageInfo.registry.get(),
+                                containerImageInfo.username.get(), containerImageInfo.password.get())));
+            } else {
+                LOG.warn("An image registry has been configured, but no push secret or credentials.");
+                LOG.warn("Unless a push secret '" + imagePushSecret + "' is provided, the build will possibly fail!");
```

Comment:
I'm not sure the message is clear enough but maybe it's just that I'm not accustomed to this.
Because for me there are two ways to fix this:
- provide a push secret in the config
- provide credentials

I don't see why pointing the user to `imagePushSecret` which will be `applicationInfo.getName() + "-push-secret"` will help fixing the issue?

Reply:
@gsmet: Thanks for the feedback. 

I think that you are right, it just overcomplicates things. We would be better of without the seconds warning.

Reply:
Updated.